build/scripts/k8s-export-openapi/main.go

```diff
@@ -0,0 +1,470 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main implements a program to convert from export-openapi.sh to Go script.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/itchyny/json2yaml"
+)
+
+func main() {
+	tmpDir := "../../tmp"
+
+	if _, err := os.Stat(tmpDir); err == nil {
+		err = os.RemoveAll(tmpDir)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	err := os.MkdirAll(tmpDir, os.ModePerm)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Start the kubectl proxy.
+	cmd := exec.Command("kubectl", "proxy")
```

Comment:
So we're running `kubectl proxy` here... so it should be started.

We might want to print/log the output from the command, to make sure it's doing what we expect.

(this might need to be put in a Go routine, since calling any of the output commands such as: https://pkg.go.dev/os/exec#Cmd.Output will block).

Reply:
Done.

Reply:
Can he be implemented using client-go here?

Reply:
Goroutine encountered an error: `Failed to start kubectl proxy: signal: interrupt`. I've excluded it.

I used the same target, `gen-embedded-openapi`, and it successfully executed using the Go script with the `-mod=mod` option. No modifications were needed in the Kubernetes checklist.

Pending task:
Check `nullable=true` missing in yaml file